### PR TITLE
🐛 Fix LGTM alerts in AMP codebase

### PR DESCRIPTION
**Pull request by rsimha**
This PR fixes a few of the security, correctness, and maintainability [issues](https://lgtm.com/projects/g/ampproject/amphtml/alerts/?mode=tree) in our code detected via static analysis from LGTM.com.

**Fixed:**
- [3p/embedly.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/3p/embedly.js?sort=name&dir=ASC&mode=heatmap#x4eccbef23f39f97c:1)
- [3p/github.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/3p/github.js#x9c3fcb1368a63323:1)
- [3p/nameframe.max.html](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/3p/nameframe.max.html#x8c1e62f82fa99681:1)
- [extensions/amp-vimeo/0.1/amp-vimeo.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/extensions/amp-vimeo/0.1/amp-vimeo.js?sort=name&dir=ASC&mode=heatmap#x118e28185bfc7f0a:1)
- [src/3p-frame.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/src/3p-frame.js?sort=name&dir=ASC&mode=heatmap#x4b4c225eccd29b3f:1)
- [validator/build.py](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/f88498129b594344e23b7193a41edda96dafc945/files/validator/build.py?sort=name&dir=ASC&mode=heatmap#xfbcf10b628aab58:1)
- [validator/engine/json-testutil.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/validator/engine/json-testutil.js?sort=name&dir=ASC&mode=heatmap#x2991175f643bd8eb:1)
- [validator/webui/build.py](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/f88498129b594344e23b7193a41edda96dafc945/files/validator/webui/build.py?sort=name&dir=ASC&mode=heatmap#xbfc2744b24bcceb4:1)
- [validator/webui/build.py](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/f88498129b594344e23b7193a41edda96dafc945/files/validator/webui/build.py?sort=name&dir=ASC&mode=heatmap#x3b3803fc6c3ab3f6:1)

**Left alone:**
- [validator/chromeextension/content_script.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/validator/chromeextension/content_script.js?sort=name&dir=ASC&mode=heatmap#x5cf43fd482805b6d:1)
- [validator/engine/validator-in-browser.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/validator/engine/validator-in-browser.js?sort=name&dir=ASC&mode=heatmap#xe6fb99b4b08ef9d2:1)

### CODE REVIEW

File: 3p/nameframe.max.html [hunk 1]
```diff
@@ -5,14 +5,14 @@
   <script>
     try {
       var name_data = JSON.parse(window.name);
-      if (!name_data || !name_data['creative']) {
+      if (!name_data || !name_data.creative) {
```
Comment (jridgewell) [inline 3p/nameframe.max.html]: Why these changes?

File: src/3p-frame.js [hunk 2]
```diff
@@ -291,7 +291,7 @@ export function getDevelopmentBootstrapBaseUrl(parentWindow, srcFileBasename) {
  */
 function getAdsLocalhost(win) {
   let adsUrl = urls.thirdParty; // local dev with a non-localhost server
-  if (adsUrl.indexOf('ampproject.net') > -1) {
+  if (adsUrl.startsWith('https://3p.ampproject.net')) {
```
Comment (jridgewell) [inline src/3p-frame.js]: I don't think this is correct, the domain uses a random subdomain every request. Eg, `d12342.3p.ampproject.net`

File: validator/engine/validator-in-browser.js [hunk 3]
```diff
@@ -51,8 +51,8 @@ function getUrl(url) {
  */
 amp.validator.isAmpCacheUrl = function(url) {
   return (
-    url.toLowerCase().indexOf('cdn.ampproject.org') !== -1 ||
```
Comment (jridgewell) [inline validator/engine/validator-in-browser.js]: Why these changes?

File: src/3p-frame.js (hunk 2 above)
Comment (jridgewell) [reply src/3p-frame.js]: This one is a breaking change.

File: 3p/nameframe.max.html (hunk 1 above)
Comment (jridgewell) [reply 3p/nameframe.max.html]: This warning isn't related to using computed property keys, it's passing something to `document.write`. Nothing we can do about that.

File: 3p/github.js [hunk 4]
```diff
@@ -26,11 +26,11 @@ import {writeScript} from './3p';
  *
  * @param {!Window} global
  * @param {string} scriptSource The source of the script, different for post and comment embeds.
- * @param {function(*)} cb
+ * @param {function()} cb
  */
 function getGistJs(global, scriptSource, cb) {
   writeScript(global, scriptSource, function() {
-    cb(global.gist);
+    cb();
```
Comment (kristoferbaxter) [inline 3p/github.js]: Are we sure the callback will work without the argument passed? This seems like a change that would have an impact on the `cb` method.

Comment (aghassemi) [reply 3p/github.js]: I double checked the callbacks for `gist` and `embedly` changes, params were unused

File: src/3p-frame.js (hunk 2 above)
Comment (rsimha) [reply src/3p-frame.js]: I wrote this line a year ago to enable local testing. In this case, the URL doesn't have a subdomain.

https://github.com/ampproject/amphtml/blob/c4a663d0ba13d0488c6fe73c55dc8c971ac6ec0d/src/config.js#L37-L38

File: 3p/github.js (hunk 4 above)
Comment (kristoferbaxter) [reply 3p/github.js]: Perfect! Will mark this as resolved.

Comment (rsimha) [reply 3p/github.js]: Yes, the alert wouldn't have fired if the params were used. The call site can is further down in the same file.

https://github.com/ampproject/amphtml/blob/c4a663d0ba13d0488c6fe73c55dc8c971ac6ec0d/3p/github.js#L55

File: validator/engine/validator-in-browser.js (hunk 3 above)
Comment (rsimha) [reply validator/engine/validator-in-browser.js]: After looking at #8424, I've reverted this change. We're going to have to ignore this alert unless there's a safer way of comparing the trailing part of a hostname.

/cc @honeybadgerdontcare

**Comment by rsimha** [comment]
I've reverted the URL comparison changes in `validator/`, since we would like to support detect URLs like `foo-com.cdn.ampproject.org` as valid AMP cache URLs. 

We'll have to come up with a different way of addressing these two alerts:
- [validator/chromeextension/content_script.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/validator/chromeextension/content_script.js?sort=name&dir=ASC&mode=heatmap#x5cf43fd482805b6d:1)
- [validator/engine/validator-in-browser.js](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/validator/engine/validator-in-browser.js?sort=name&dir=ASC&mode=heatmap#xe6fb99b4b08ef9d2:1)

/cc @honeybadgerdontcare @Gregable

File: 3p/nameframe.max.html (hunk 1 above)
Comment (rsimha) [reply 3p/nameframe.max.html]: There are two warnings in this file. This change addresses one of them. You're right about the other one. See [3p/nameframe.max.html](https://lgtm.com/projects/g/ampproject/amphtml/snapshot/62a6f805a2d91db98d7bd8751b61b9171d0a8cc5/files/3p/nameframe.max.html#x8c1e62f82fa99681:1)

File: src/3p-frame.js (hunk 2 above)
Comment (rsimha) [reply src/3p-frame.js]: Updated this to be a simple equality check, since this is merely looking for an overridden flag during local testing.

**Comment by rsimha** [comment]
I'll merge this after the one last JS [change](https://github.com/ampproject/amphtml/pull/22488/files#diff-461b6ee8143fcc21751646dd5ef2fcfbR101) in `validator/` is reviewed.

**Comment by rsimha** [comment]
Validator checks are [green](https://travis-ci.org/ampproject/amphtml/jobs/536926856). Merging this.
